base/shared/src/main/scala/scalaz/ct/phantom.scala

```diff
@@ -6,7 +6,10 @@ import scalaz.data.Const
 import scala.language.experimental.macros
 
 trait PhantomClass[F[_]] extends FunctorClass[F] with ContravariantClass[F] {
+
   def pmap[A, B](ma: F[A]): F[B]
+
+  override def imap[A, B](ma: F[A])(f: A => B)(g: B => A): F[B] = pmap(ma)
 }
```

Comment:
arguably you can also add overrides for `map` and `contramap` but that doesn't need to be done now.

Reply:
Would that make `DeriveMapContramap` redundant?

Reply:
ah, yeah, we're doing it there. I'm working on removing the `Derive*` traits, so don't worry.